haddock-api/src/Haddock/Backends/Xhtml/Decl.hs

```diff
@@ -634,9 +634,9 @@ ppInstances links origin instances splice unicode pkg qual
   -- force Splice = True to use line URLs
   where
     instName = getOccString origin
-    instDecl :: Int -> DocInstance DocNameI -> (SubDecl, Maybe Module, Located DocName)
+    instDecl :: Int -> DocInstance DocNameI -> (String, SubDecl, Maybe Module, Located DocName)
     instDecl no (inst, mdoc, loc, mdl) =
-        ((ppInstHead links splice unicode qual mdoc origin False no inst mdl), mdl, loc)
+        (getOccString (ihdClsName inst) <> "_" <> show no <> ":", (ppInstHead links splice unicode qual mdoc origin False no inst mdl), mdl, loc)
```

Comment:
I think defining the added `String` and the `Maybe Module` in separate in a where-clause would help with readability.